app/static/js/admin/event/wizard/vue-app.js

```diff
@@ -155,6 +155,9 @@ var app = new Vue({
         },
         'step': function (step) {
             history.replaceState(null, '', "./" + step);
+        },
+        'event.timezone': function (value) {
+            this.event.timezone = value;
```

Comment:
I think you mis-understood how vue works. This `event.timezone` watcher itself is invoked when `this.event.timezone` changes. The watcher is unnecessary. `this.event.timezone` is automatically updated by vue.